scripts/skills/perks/perk_rf_king_of_all_weapons.nut

```diff
@@ -16,22 +16,39 @@ this.perk_rf_king_of_all_weapons <- ::inherit("scripts/skills/skill", {
 
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
-		if (!this.isSkillValid(_skill))
+		if (_targetEntity == null || !this.isSkillValid(_skill))
 			return;
 
-		if (_targetEntity != null && ::Math.rand(1, 100) <= this.getChance())
+		local headArmor = _targetEntity.getArmor(::Const.BodyPart.Head);
+		local bodyArmor = _targetEntity.getArmor(::Const.BodyPart.Body);
+		local isHeadLower = headArmor < bodyArmor;
+		local isBodyLower = bodyArmor < headArmor;
+
+		local chance = this.getChance();
+		local actor = this.getContainer().getActor();
+		if (actor.isPreviewing())
+		{
+			chance *= 0.01;
+			if (isHeadLower)
+			{
+				_properties.HitChance[::Const.BodyPart.Head] = 100 * (chance + (1.0 - chance) * _properties.HitChance[::Const.BodyPart.Head] * 0.01);
```

Comment:
I dont understand what is happening in this line. Too many numbers and operators.

The second line below for `isBodyLower` I can parse in my head. That one scales down the headshot chance harder, the higher our chance is to auto-hit the body via this perks effect